### PR TITLE
26 reconstruct cinema images

### LINKED ISSUE #26: Reconstruct CineMA images
After n epochs of training, reconstruct the masked patches for CineMA to judge the performance visually.

### PR DESCRIPTION
**Pull request by LevanBokeria**
# Pull request overview

This pull request implements image reconstruction functionality for the CineMA masked autoencoder (MAE) pretraining pipeline, enabling visualization of masked and reconstructed cardiac MRI images during training. 
It also updates the dataloader sampler from sequential to random.

# Changes:

- Adds new utility script for MAE reconstruction and visualization (mae_recon.py), along with its tests.
- Integrates reconstruction visualization into the CineMA pretraining pipeline with configurable save intervals
- Removes the deprecated vit_pretrain.py file and renames vit_pretrain_hydra.py to vit_pretrain.py
- Updates configuration files to support reconstruction saving frequency
- Updates the CineMA dataloader to use a random sampler instead of the sequential sampler.

### CODE REVIEW

File: heartfm/cinema_scripts/cinema_pretrain.py [hunk 1]
```diff
@@ -369,10 +414,41 @@ def evaluate_one_epoch(
...
+                    enc_mask_dict=mask_recon,
+                    sax_slices=sax_slices,
+                    filepath=filepath,
+                    fig_suptitle=fig_suptitle,
+                )
+
+                # Send figure to TensorBoard
+                writer.add_figure(
+                    f"Reconstruction/test_epoch_{epoch}", fig, global_step=epoch
+                )
```
[the remark this answers is not in the document]
Comment (LevanBokeria) [reply heartfm/cinema_scripts/cinema_pretrain.py:446]: @copilot how would I close the figure?

File: heartfm/cinema_scripts/utils/mae_recon.py
Comment (crangelsmith) [inline heartfm/cinema_scripts/utils/mae_recon.py:1]: Is this compatible with the ViT run as well? Can we use it in both?

**Review by crangelsmith — APPROVED** [review]

File: heartfm/cinema_scripts/utils/mae_recon.py
Comment (LevanBokeria) [reply heartfm/cinema_scripts/utils/mae_recon.py:1]: Not as of now, but could consider writing a general image recon script for different models. Although, its possible that the models are just too different in terms of how they structure the data and thus what sort of inputs they need, and building a universal script will be futile.
